CRM/Contact/BAO/Relationship.php

```diff
@@ -675,7 +675,13 @@ public static function clearCurrentEmployer($id, $action) {
         $employerRelTypeId = CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_RelationshipType', 'Employee of', 'id', 'name_a_b');
 
         if ($relationship->relationship_type_id == $employerRelTypeId && $relationship->contact_id_b == $sharedContact->employer_id) {
-          CRM_Contact_BAO_Contact_Utils::clearCurrentEmployer($relationship->contact_id_a);
+          $currentEmployerRlationship = new CRM_Contact_DAO_Relationship();
+          $currentEmployerRlationship->contact_id_a = $relationship->contact_id_a;
+          $currentEmployerRlationship->end_date=NULL;
+          $currentEmployerRlationship->is_active=1;
+          if(!$currentEmployerRlationship->find(TRUE)){
+	     CRM_Contact_BAO_Contact_Utils::clearCurrentEmployer($relationship->contact_id_a);
+          }
```

Comment:
I think the above checking can be done by an existing function in this file. @anaskoara Can you check if you can achieve the same outcome by using the if condition as shown below?

    $existingCurrentEmployerRelationship = self::getRelationship($relationship->contact_id_a, self::CURRENT, 0, 0, 0, NULL, NULL, FALSE, array('relationship_type_id' => $employerRelTypeId));

    if (!empty($existingCurrentEmployerRelationship)) {
      //there is an active relationship, so avoid clearing the employer
    }